Mathlib/MeasureTheory/Decomposition/Lebesgue.lean

```diff
@@ -99,24 +91,64 @@ theorem haveLebesgueDecomposition_spec (μ ν : Measure α) [h : HaveLebesgueDec
   exact Classical.choose_spec h.lebesgue_decomposition
 #align measure_theory.measure.have_lebesgue_decomposition_spec MeasureTheory.Measure.haveLebesgueDecomposition_spec
 
-instance instHaveLebesgueDecomposition_zero_left : HaveLebesgueDecomposition 0 ν where
-  lebesgue_decomposition := ⟨⟨0, 0⟩, measurable_zero, MutuallySingular.zero_left, by simp⟩
+lemma rnDeriv_of_not_haveLebesgueDecomposition (h : ¬ HaveLebesgueDecomposition μ ν) :
+    μ.rnDeriv ν = 0 := by
+  rw [rnDeriv, dif_neg h]
 
-instance instHaveLebesgueDecomposition_zero_right : HaveLebesgueDecomposition μ 0 where
-  lebesgue_decomposition := ⟨⟨μ, 0⟩, measurable_zero, MutuallySingular.zero_right, by simp⟩
+lemma singularPart_of_not_haveLebesgueDecomposition (h : ¬ HaveLebesgueDecomposition μ ν) :
+    μ.singularPart ν = 0 := by
+  rw [singularPart, dif_neg h]
+
+@[measurability]
+theorem measurable_rnDeriv (μ ν : Measure α) : Measurable <| μ.rnDeriv ν := by
+  by_cases h : HaveLebesgueDecomposition μ ν
+  · exact (haveLebesgueDecomposition_spec μ ν).1
+  · rw [rnDeriv_of_not_haveLebesgueDecomposition h]
+    exact measurable_zero
+#align measure_theory.measure.measurable_rn_deriv MeasureTheory.Measure.measurable_rnDeriv
+
+theorem mutuallySingular_singularPart (μ ν : Measure α) : μ.singularPart ν ⟂ₘ ν := by
+  by_cases h : HaveLebesgueDecomposition μ ν
+  · exact (haveLebesgueDecomposition_spec μ ν).2.1
+  · rw [singularPart_of_not_haveLebesgueDecomposition h]
+    exact MutuallySingular.zero_left
+#align measure_theory.measure.mutually_singular_singular_part MeasureTheory.Measure.mutuallySingular_singularPart
 
 theorem haveLebesgueDecomposition_add (μ ν : Measure α) [HaveLebesgueDecomposition μ ν] :
     μ = μ.singularPart ν + ν.withDensity (μ.rnDeriv ν) :=
   (haveLebesgueDecomposition_spec μ ν).2.2
 #align measure_theory.measure.have_lebesgue_decomposition_add MeasureTheory.Measure.haveLebesgueDecomposition_add
 
-instance haveLebesgueDecomposition_smul (μ ν : Measure α) [HaveLebesgueDecomposition μ ν]
-    (r : ℝ≥0) : (r • μ).HaveLebesgueDecomposition ν where
+lemma singularPart_add_rnDeriv (μ ν : Measure α) [HaveLebesgueDecomposition μ ν] :
+    μ.singularPart ν + ν.withDensity (μ.rnDeriv ν) = μ := (haveLebesgueDecomposition_add μ ν).symm
+
+lemma rnDeriv_add_singularPart (μ ν : Measure α) [HaveLebesgueDecomposition μ ν] :
+    ν.withDensity (μ.rnDeriv ν) + μ.singularPart ν = μ := by rw [add_comm, singularPart_add_rnDeriv]
+
+end ByDefinition
+
+section HaveLebesgueDecomposition
+
+instance instHaveLebesgueDecomposition_zero_left : HaveLebesgueDecomposition 0 ν where
```

Comment:
instance names should not contain underscores, if I understand correctly. So
```suggestion
instance instHaveLebesgueDecompositionZeroLeft : HaveLebesgueDecomposition 0 ν where
```
Same thing below.